lib/app/modules/addOrUpdateAlarm/views/add_or_update_alarm_view.dart

```diff
@@ -122,633 +125,717 @@ class AddOrUpdateAlarmView extends GetView<AddOrUpdateAlarmController> {
-                  controller.mutexLock.value == true)
+              controller.mutexLock.value == true)
               ? const SizedBox()
               : Padding(
-                  padding: const EdgeInsets.all(18.0),
-                  child: SizedBox(
-                    height: height * 0.06,
-                    width: width * 0.8,
-                    child: TextButton(
-                      style: ButtonStyle(
-                        backgroundColor:
-                            MaterialStateProperty.all(kprimaryColor),
+            padding: const EdgeInsets.all(18.0),
+            child: SizedBox(
+              height: height * 0.06,
+              width: width * 0.8,
+              child: TextButton(
+                style: ButtonStyle(
+                  backgroundColor:
+                  MaterialStateProperty.all(kprimaryColor),
+                ),
+                child: Text(
+                  (controller.alarmRecord == null) ? 'Save' : 'Update',
+                  style:
+                  Theme.of(context).textTheme.displaySmall!.copyWith(
+                    color: themeController.isLightMode.value
+                        ? kLightPrimaryTextColor
+                        : ksecondaryTextColor,
+                  ),
+                ),
+                onPressed: () async {
+                  Utils.hapticFeedback();
+                  if (controller.userModel != null) {
+                    controller.offsetDetails[controller.userModel!.id] = {
+                      'offsettedTime': Utils.timeOfDayToString(
+                        TimeOfDay.fromDateTime(
+                          Utils.calculateOffsetAlarmTime(
+                            controller.selectedTime.value,
+                            controller.isOffsetBefore.value,
+                            controller.offsetDuration.value,
+                          ),
+                        ),
                       ),
-                      child: Text(
-                        (controller.alarmRecord == null) ? 'Save' : 'Update',
-                        style:
-                            Theme.of(context).textTheme.displaySmall!.copyWith(
-                                  color: themeController.isLightMode.value
-                                      ? kLightPrimaryTextColor
-                                      : ksecondaryTextColor,
-                                ),
+                      'offsetDuration': controller.offsetDuration.value,
+                      'isOffsetBefore': controller.isOffsetBefore.value,
+                    };
+                  } else {
+                    controller.offsetDetails.value = {};
+                  }
+
+                  String time = Utils.timeOfDayToString(
+                    TimeOfDay.fromDateTime(
+                      controller.selectedTime.value,
+                    ),
+                  );
+                  bool res = await IsarDb.doesAlarmExistTime(time,true);
```

Comment:
Also, this doesn't account for shared alarms right? I could have an alarm in firestore and another in isar with the same time, that would have to be fixed as well.

Reply:
ok I will fix it